plugins/output/adapt/importsource.js

```diff
@@ -462,114 +452,132 @@ function ImportSource(req, done) {
-              var folderMatch = "(course\/)((\\w){2}\/)" + '(' + assetFolders[index] + ')';
-              var assetFolderRegex = new RegExp(folderMatch, "gi");
+          for (index = 0; index < assetFolders.length; ++index) {
+            var folderMatch = "(course\/)((\\w){2}\/)" + '(' + assetFolders[index] + ')';
+            var assetFolderRegex = new RegExp(folderMatch, "gi");
 
-              if (!dirName.match(assetFolderRegex)) continue;
-              newDirName = dirName.replace(assetFolderRegex, newAssetPath);
-            }
+            if (!dirName.match(assetFolderRegex)) continue;
+            newDirName = dirName.replace(assetFolderRegex, newAssetPath);
+          }
 
-            var fileExt = path.extname(value);
-            var fileName = path.basename(value);
-            if (newDirName && fileName && fileExt) {
-              try {
-                var fileId = metadata.idMap[fileName];
-                var newFileName = metadata.assetNameMap[fileId];
-                if (newFileName) {
-                  this.update(newDirName + '/' + newFileName); // always use '/' for paths in content
-                }
-              } catch (e) {
-                logger.log('error', e);
-                return;
+          var fileExt = path.extname(value);
+          var fileName = path.basename(value);
+          if (newDirName && fileName && fileExt) {
+            try {
+              var fileId = metadata.idMap[fileName];
+              var newFileName = metadata.assetNameMap[fileId];
+              if (newFileName) {
+                this.update(newDirName + '/' + newFileName); // always use '/' for paths in content
               }
+            } catch (e) {
+              logger.log('error', e);
+              return;
             }
-          });
-          cb();
-        }
+          }
+        });
+        cb();
+      }
     ], function(error, results) {
       if(error) return done(error);
-
-      if (detachedElementsMap[originalData._id]) {
-        // do not import detached elements 
+      if (detachedElementsMap[originalData._id]) { // do not import detached elements
         return done();
-      }
-
-      // now we're ready to create the content
+      } // now we're ready to create the content
       app.contentmanager.getContentPlugin(type, function(error, plugin) {
         if(error) return done(error);
         plugin.create(data, function(error, record) {
           if(error) {
             logger.log('warn', 'Failed to import ' + type + ' ' + (originalData._id || '') + ' ' + error);
             return done();
-          }
-          // Create a courseAssets record if needed
-          createCourseAssets(type, record, function(error){
-            if(error) return done(error, record);
-            done(null, record);
-          });
+          } // Create a courseAssets record if needed
+          createCourseAssets(type, record, () => done(null, record));
```

Comment:
cb will never return an error. Is this intended? 
```
createCourseAssets(type, record, error => done(error, record));
```

Reply:
Yeah, the original code didn't do this -- will gladly add this if needed. 
@dancgray do you have any opinions?

Reply:
@taylortom I think it should return the error.